### PR TITLE
self-hosted runners for python actions

### PR DESCRIPTION
**Pull request by iliadrm**
added the following self-hosted runners: rpi2, rpi3, rpi4, beaglebone, jetson nano

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -59,13 +60,35 @@ jobs:
...
+    - uses: actions/checkout@v2
+
+    - name: Pre-build dependencies
+      run: python3 -m pip install --upgrade pip
+
+    - name: Install dependencies
+      run: pip3 install -r requirements.txt
+
+    - name: Test
+      run: python3 test_porcupine.py ${{secrets.PV_VALID_ACCESS_KEY}}
```
Comment (ErisMik) [inline .github/workflows/python.yml:94]: Is there any reason we can't combine the two jobs seeing as how the steps are nearly incidental? I think it'll be easier to maintain.

Comment (iliadrm) [reply .github/workflows/python.yml:94]: I reverted the extra commits.

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -44,11 +44,12 @@ defaults:
     working-directory: binding/python
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  build-git-hosted:
```
Comment (ErisMik) [inline .github/workflows/python.yml:47]: Small nitpick, can we call this `build-github-hosted`?

**Review by ErisMik — APPROVED** [review]
LGTM after small nitpick

**Pull request merged**